frontends/p4/checkCoreMethods.cpp

```diff
@@ -0,0 +1,129 @@
+/*
```

Comment:
This code used to be in the typeChecker, but was moved to this separate file.
Otherwise it is unchanged.